File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/course_detail/coures_students/presenter/StudentListPresenterImpl.java
package org.wikiedufoundation.wikiedudashboard.course_detail.coures_students.presenter;


import org.wikiedufoundation.wikiedudashboard.course_detail.coures_students.data.StudentListResponse;
import org.wikiedufoundation.wikiedudashboard.course_detail.coures_students.provider.StudentListProvider;
import org.wikiedufoundation.wikiedudashboard.course_detail.coures_students.view.StudentListView;

import org.wikiedufoundation.wikiedudashboard.helper.PresenterCallback;


public class StudentListPresenterImpl implements StudentListPresenter {
    private StudentListView studentListView;
    private StudentListProvider studentListProvider;

    public StudentListPresenterImpl(StudentListView studentListView,StudentListProvider studentListProvider) {
        this.studentListProvider=studentListProvider;
        this.studentListView=studentListView;
    }

    @Override
    public void requestStudentList(String url) {
        studentListView.showProgressBar(true);
        studentListProvider.requestStudentList(url, new PresenterCallback() {
            @Override
            public void onSuccess(Object o) {
                studentListView.showProgressBar(false);
                StudentListResponse studentListsResponse = (StudentListResponse) o;
                studentListView.setData(studentListsResponse);
            }

            @Override
            public void onFailure() {
                studentListView.showProgressBar(false);
                studentListView.showMessage("Unable to connect to server.");

            }
        });
    }




}

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/coures_students/presenter/StudentListPresenterImpl.kt
package org.wikiedufoundation.wikiedudashboard.ui.coursedetail.coures_students.presenter


import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.coures_students.data.StudentListResponse
import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.coures_students.provider.StudentListProvider
import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.coures_students.view.StudentListView

import org.wikiedufoundation.wikiedudashboard.util.PresenterCallback


class StudentListPresenterImpl(private val studentListView: StudentListView, private val studentListProvider: StudentListProvider) : StudentListPresenter {

    override fun requestStudentList(url: String) {
        studentListView.showProgressBar(true)
        studentListProvider.requestStudentList(url, object : PresenterCallback<Any> {
            override fun onSuccess(o: Any) {
                studentListView.showProgressBar(false)
                val studentListsResponse = o as StudentListResponse
                studentListView.setData(studentListsResponse)
            }

            override fun onFailure() {
                studentListView.showProgressBar(false)
                studentListView.showMessage("Unable to connect to server.")

            }
        })
    }


}
